src/ai/ai.ts

```diff
@@ -0,0 +1,172 @@
+    await getService(this.ainize!, serviceName); // NOTE(jiyoung): check if the service is deployed on Ainize.
+
+    const txBody = this.buildTxBodyForConfigureAi(objectId, serviceName, address);
+    const result = await sendTx(this.ain, txBody);
+
+    return { ...result, config: { name: serviceName } };
+  }
+
+  /**
+   * Retrieves the credit balance for a service.
+   * @param {string} serviceName - The name of Ainize service.
+   * @returns {Promise<number|null>} Returns a promise that resolves with the credit balance.
+   */
+  async getCredit(serviceName: string): Promise<number> {
+    const address = await this.ain.signer.getAddress();
+
+    let balance = 0;
+
+    if (serviceName === DEFAULT_AINIZE_SERVICE_NAME) {
+      const opType = OperationType.GET_CREDIT;
+      const body = { address };
+
+      const response = await request<number>(this.ainize!, {
+        serviceName: serviceName,
+        opType,
+        data: body,
+        timeout: 2*60*1000, // 2min
+      });
+      balance = response.data;
+    } else {
+      const service = await getService(this.ainize!, serviceName);
+      balance = await service.getCreditBalance();
+    }
+
+    return balance;
+  }
+
+  /**
+   * @todo Enable deposit function once withdrawal is fully implemented and tested.
+   * Deposits a credits for a service.
+   * @param {string} serviceName - The name of Ainize service.
+   * @param {number} amount - The amount of credits to deposit.
+   * @returns {Promise<CreditTransactionResult>} Returns a promise that resolves with the deposit transaction info.
+   */
+  /*
+  async depositCredit(serviceName: string, amount: number): Promise<CreditTransactionResult> {
+    const address = await this.ain.signer.getAddress();
+
+    const service = await getService(this.ainize!, serviceName); // NOTE(jiyoung): check if the service is deployed on Ainize.
+
+    const currentBalance = await service.getCreditBalance();
+    const txHash = await service.chargeCredit(amount);
+    const updatedBalance = await this.waitForUpdate(
+      service,
+      currentBalance + amount,
+      60*1000, // 1min
+      txHash
+    );
+
+    return { tx_hash: txHash, address, balance: updatedBalance };
+  }
+  */
+
+  // TODO(jiyoung): refactor this method.
+  async getUserTokensByStatus(
+    objectId: string,
+    address: string,
+    status?: TokenStatus,
+    { limit = 20, offset = 0, order = 'desc' }: QueryParams = {}
+  ) {
+    await validateObject(this.ain, objectId);
+
+    const appId = AinftObject.getAppId(objectId);
+    const tokensPath = Path.app(appId).tokens().value();
+    const allTokens: NftTokens = (await this.ain.db.ref(tokensPath).getValue()) || {};
+
+    const tokens = Object.entries(allTokens).reduce<NftToken[]>((acc, [id, token]) => {
+      if (token.owner === address) {
+        acc.push({ tokenId: id, ...token });
+      }
+      return acc;
+    }, []);
+
+    tokens.map((token) => {
+      let status = 'minted';
```

Comment:
how about adding an enum definition for this?

Reply:
done!